src/frontend/web_application/src/scenes/DevicesSettings/components/DeviceForm/presenter.jsx

```diff
@@ -25,16 +14,35 @@ class DeviceForm extends Component {
     i18n: PropTypes.shape({}).isRequired,
   };
 
-  state = {
-    ...generateStateFromProps(this.props),
-  };
+  static generateStateFromProps(props, prevState) {
+    const { isFetching, device } = props;
 
-  componentWillMount() {
-    this.setState(generateStateFromProps(this.props));
+    return {
+      ...prevState,
+      initialized: !isFetching && device,
+      device,
+    };
   }
 
-  componentWillReceiveProps(newProps) {
-    this.setState(generateStateFromProps(newProps));
+  static initialState = {
+    initialized: false,
+    device: {
+      name: '',
+      type: '',
+      // locations: [],
+    },
+  };
+
+  state = this.constructor.generateStateFromProps(this.props, this.constructor.initialState);
+
+  componentDidUpdate(prevProps) {
+    const propNames = ['device'];
+    const hasChanged = propNames.some(propName => this.props[propName] !== prevProps[propName]);
+
+    if (!this.state.initialized && hasChanged) {
+      // eslint-disable-next-line react/no-did-update-set-state
```

Comment:
😱

We will have to remove a lot of these…

Reply:
it's a good reminder to check  what is the best practice